runtime/tests/interpreter/interpreter_test.go

```diff
@@ -9435,3 +9435,58 @@ func TestHostFunctionStaticType(t *testing.T) {
 		assert.Equal(t, xValue.StaticType(), yValue.StaticType())
 	})
 }
+
+func TestInterpretArrayTypeInference(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("anystruct with empty array", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            fun test(): Type {
+                let x: AnyStruct = []
+                return x.getType()
+            }
+        `)
+
+		value, err := inter.Invoke("test")
+		require.NoError(t, err)
+
+		AssertValuesEqual(
+			t,
+			inter,
+			interpreter.TypeValue{
+				Type: interpreter.VariableSizedStaticType{
+					Type: interpreter.PrimitiveStaticTypeAnyStruct,
+				},
+			},
+			value,
+		)
+	})
+
+	t.Run("anystruct with numeric array", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            fun test(): Type {
+                let x: AnyStruct = [1, 2, 3]
+                return x.getType()
+            }
+        `)
+
+		value, err := inter.Invoke("test")
+		require.NoError(t, err)
+
+		AssertValuesEqual(
+			t,
+			inter,
+			interpreter.TypeValue{
+				Type: interpreter.VariableSizedStaticType{
+					Type: interpreter.PrimitiveStaticTypeAnyStruct,
```

Comment:
Totally fine and not a problem, but this behaviour is now different from the current one, where the type of the first element was used, so would have been `Int` here

Reply:
Yeah, I also don't have a strong case against the old behavior. Only thing was that the old behavior is not consistent with `let x: AnyStruct = []`, where the array is empty and the type cannot be inferred by itself.

I guess it's fine to handle the two cases separately. WIll revert back to the old behavior

Reply:
Updated in 67e3268522d210d752cc49c0a34613fda665826b

Reply:
Great, thank you!